src/mscorlib/src/System/TimeSpan.cs

```diff
@@ -405,6 +405,42 @@ public String ToString(String format, IFormatProvider formatProvider)
             return t1.Add(t2);
         }
 
+        public static TimeSpan operator *(TimeSpan timeSpan, double factor)
+        {
+            if (double.IsNaN(factor))
+            {
+                throw new ArgumentException(Environment.GetResourceString("Arg_CannotBeNaN"), nameof(factor));
+            }
+
+            double ticks = timeSpan.Ticks * factor;
+            if (ticks > long.MaxValue | ticks < long.MinValue)
+            {
+                throw new OverflowException(Environment.GetResourceString("Overflow_TimeSpanTooLong"));
+            }
+
+            return FromTicks((long)ticks);
```

Comment:
Right you are. That seems probably too coarse for this, in that `FromTicks(2) * 2.0` should probably work as one would expect.